src/poetry/core/packages/specification.py

```diff
@@ -71,9 +71,15 @@ def source_subdirectory(self) -> str | None:
     def features(self) -> frozenset[str]:
         return self._features
 
-    def is_same_package_as(self, other: PackageSpecification) -> bool:
-        if other.complete_name != self.complete_name:
-            return False
+    def is_same_package_as(
+        self, other: PackageSpecification, *, allow_similar: bool = False
+    ) -> bool:
+        if allow_similar:
```

Comment:
Should we also treat `foo (1.2.3 git rev)` as being similar to `foo (1.2.3)` ?

Reply:
It depends:

I suppose for the current downstream use case that would not be suitable because "similar" means same package with any additional features. That can be true for `foo (1.2.3 git rev)` and `foo (1.2.3)` but is not safe.

Considering other use cases (I don't know yet), it might be suitable.

If we only want to map the current use case, maybe, the parameter should be renamed: `match_complete_name=True` or `ignore_features=False`? (I think the latter would require a slightly different implementation but is most exactly what is required by the current downstream use case.)

If we don't want to map the current use case, we can keep the parameter name and treat your example as similar.

Reply:
I thought about it again and I changed it to `ignore_features`. That is better because it is clear and exactly what is required. `allow_similar` is too ambiguous and open to interpretation. I think I will revisit the downstream usage of `allow_similar` after this has been merged and included in the next (pre-)release.

Reply:
So there is another issue that I keep forgetting about. There is a use case of "overrides" in development.

```sh
poetry add project
poetry add --group=dev git+https://github.com/org/project.git
# alternatively
poetry add --group=dev --editable ./project
```

I am not sure how the recent changes have affected this use case.

Also, along the same lines (maybe) https://github.com/python-poetry/poetry/issues/5617.

Reply:
Althought, it is important to note that these are poetry specific use case and might not be interesting for core itself.

Reply:
I agree. Lets drop this for now.